Sources/Purchasing/StoreKit2/StoreKit2StorefrontListener.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  StoreKit2StorefrontListener.swift
+//
+//  Created by Juanpe Catalán on 5/5/22.
+
+import Foundation
+import StoreKit
+
+protocol StoreKit2StorefrontListenerDelegate: AnyObject {
+    func storefrontDidUpdate()
+}
+
+@available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+class StoreKit2StorefrontListener {
+
+    private(set) var taskHandle: Task<Void, Never>?
```

Comment:
I would implement `didSet` here like so:
```swift
didSet {
  if self.taskHandle !== oldValue {
    oldValue?.cancel()
  }
}
```

Then you only need to set it to `nil` or the new value.

Reply:
much better 👍🏼 Thanks :)

Reply:
Just a question, why do you have to check if they are identical, checking if they are not `equal` wouldn't be enough? 🤔

Reply:
Done! ✅  [cee546e](https://github.com/RevenueCat/purchases-ios/pull/1557/commits/cee546efcbb20d7163454f5a65a050db76ece160)